pkg/blobinfocache/sqlite/sqlite.go

```diff
@@ -430,7 +430,19 @@ func (sqc *cache) RecordDigestCompressorName(anyDigest digest.Digest, compressor
 // appendReplacementCandidates creates prioritize.CandidateWithTime values for (transport, scope, digest), and returns the result of appending them to candidates.
 func (sqc *cache) appendReplacementCandidates(candidates []prioritize.CandidateWithTime, tx *sql.Tx, transport types.ImageTransport, scope types.BICTransportScope, digest digest.Digest, requireCompressionInfo bool) ([]prioritize.CandidateWithTime, error) {
 	var rows *sql.Rows
+	res := []prioritize.CandidateWithTime{}
 	var err error
+	compressorName := blobinfocache.UnknownCompression
+	rows, err = tx.Query("SELECT compressor FROM DigestCompressors WHERE digest = ?", digest.String())
```

Comment:
I rather suspect that this extra query is unnecessary; it could be combined into the `JOIN`s below. (But I didn’t prototype that, it might be too ugly to do.

---

At the very least, if this `compressorName` is _only_ used if the with-location query below finds nothing, it should only be executed in that case (down in the newly added `if`).